dev/com.ibm.ws.security.wim.adapter.ldap_fat.krb5.1/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/krb5/LdapContainer.java

```diff
@@ -0,0 +1,137 @@
+
+    protected static String bindPrincipalName = LdapKerberosUtils.BIND_PRINCIPAL_NAME; // default, override in extending class
+
+    // NOTE: If this is ever updated, don't forget to push to docker hub, but DO NOT overwrite existing versions
+    private static final String IMAGE = "zachhein/ldap-server:0.3";
+
+    private int tcp_389;
+
+    public LdapContainer(Network network) {
+        super(IMAGE);
+        withNetwork(network);
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    protected void configure() {
+        withNetworkAliases(LDAP_HOSTNAME);
+        withCreateContainerCmdModifier(cmd -> {
+            cmd.withHostName(LDAP_HOSTNAME);
+        });
+        withEnv("KRB5_REALM", KRB5_REALM);
+        Log.info(c, "configure", "setting env LDAP_HOSTNAME: " + LDAP_HOSTNAME);
+        withEnv("LDAP_HOSTNAME", LDAP_HOSTNAME);
+        withEnv("KRB5_PASS", KRB5_PASS);
+        withEnv("KDC_HOSTNAME", KDC_HOSTNAME);
+        Log.info(c, "configure", "setting env KDC_HOSTNAME: " + KDC_HOSTNAME);
+
+        withLogConsumer(new SimpleLogConsumer(c, "ldap"));
+        waitingFor(new LogMessageWaitStrategy()
+                        .withRegEx("^.*stop the ldap.*$")
+                        .withStartupTimeout(Duration.ofSeconds(FATRunner.FAT_TEST_LOCALRUN ? 50 : 300)));
+        withCreateContainerCmdModifier(cmd -> {
+            //Add previously exposed ports and UDP port
+
+            List<ExposedPort> exposedPorts = new ArrayList<ExposedPort>();
+            for (ExposedPort p : cmd.getExposedPorts()) {
+                Log.info(c, "configure", "ExposedPort=" + p.getPort());
+                exposedPorts.add(p);
+            }
+            exposedPorts.add(ExposedPort.tcp(389));
+            cmd.withExposedPorts(exposedPorts);
+
+            // Add previous port bindings and KDC and LDAP ports
+            Ports ports = cmd.getPortBindings();
+
+            String ldapPortMapping = String.format("%d:%d/%s", 389, 389, InternetProtocol.TCP);
+            Log.info(c, "configure", "adding ldap port mapping: " + ldapPortMapping);
+
+            Log.info(c, "configure", "PortBinding.parse(ldapPortMapping): " + PortBinding.parse(ldapPortMapping));
+            ports.add(PortBinding.parse(ldapPortMapping));
+
+            Log.info(c, "configure", "ports: " + ports);
+            cmd.withPortBindings(ports);
+            cmd.withHostName(LDAP_HOSTNAME);
+        });
+    }
+
+    //@Override
+    //protected void containerIsStarted(InspectContainerResponse containerInfo) {
+    ///    String tcp389 = containerInfo.getNetworkSettings().getPorts().getBindings().get(new ExposedPort(389, InternetProtocol.TCP))[0].getHostPortSpec();
+    //    tcp_389 = Integer.valueOf(tcp389);
+    // }
+
+    @Override
+    public void start() {
+        String dockerHostIp = DockerClientFactory.instance().dockerHostIpAddress();
+        EXTERNAL_HOSTNAME = FATSuite.dockerHostname;
+        withEnv("EXTERNAL_HOSTNAME", EXTERNAL_HOSTNAME);
```

Comment:
Same comment here... this should be part of the configure set, and there shouldn't be tight coupling between FATSuite and the container.